General/OOD/SemanticConstructorOverloading/SemanticConstructorOverloading.Problem/Optional Constructor Parameter Collision/Bad/SampleClass.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+
+namespace SemanticConstructorOverloading.Problem.Optional_Constructor_Parameter_Collision.Bad
+{
+    /// <summary>
+    /// Optional Constructor parameters cause problems when they are of the same data type, especially when
+    /// we need to be able to construct the class properly when either of the parameters of the common type are
+    /// passed to the constructor. This is because C# class constructors can only have a single instance of
+    /// a constructor overload for a given input type signature.
+    /// </summary>
+    public class SampleClass
+    {
+        // Public property assignments managed by constructor. In this use case, these will most
+        // likely be used for some kind of object configuration option.
+        public string StringValueA { get; set; }
+        public string StringValueB { get; set; }
+
+
+        /// <summary>
+        /// When two parameters of the same type are OPTIONAL in a constructor, specifying them as 
+        /// optional parameters is easier to code and generally better than specifying all
+        /// possible combinations of non-optional parameters. This is still an ugly solution, but
+        /// it's better than multiple overloaded constructors with parsing required to differentiate 
+        /// parameter values and assign to the correct value.
+        /// </summary>
+
+        /// <summary>
+        /// Trivial case
+        /// </summary>
+        public SampleClass()
+        {
+            StringValueA = "Lukewarm";
+            StringValueB = "Computer";
+        }
+
+        /// <summary>
+        /// Trivial case
+        /// <param name="parameterA"></param>
+        /// <param name="parameterB"></param>
+        /// </summary>
+        public SampleClass(string parameterA, string parameterB)
+        {
+            StringValueA = parameterA;
+            StringValueB = parameterB;
+        }
+
+
+        /// <summary>
+        /// Only a single constructor with a single string parameter can be created. So we need to inspect
+        /// the values provided to see which string to assign to, and apply default values to the other.
+        /// <param name="someValue"></param>
+        /// </summary>
+        public SampleClass(string someValue)
+        {
+            // We have no idea whether we're getting a parameterA argument or a parameterB argument
+            // This means we have to know the range of values that are valid for each and assign to the
+            // correct property. We also have to reject values that don't match either range.
+            // I.e. We are forced to perform validation in the constructor.
+            string[] ParameterAValues = { "Lukewarm", "Hot", "Cold" };
```

Comment:
This is bad code, but for different reasons. 

If we have an object with A and B properties, based on our business case it can be valid not setting the other one. However, based on name, it should be clear what property will get the value set.

For example, just because property A, B, C and D all exist, setting A, B, C shouldn't be impossible or even complicated, using arbitrary names.